common/include/pcl/PCLPointCloud2.h

```
@@ -49,6 +51,51 @@ namespace pcl
   public:
     using Ptr = boost::shared_ptr< ::pcl::PCLPointCloud2>;
     using ConstPtr = boost::shared_ptr<const ::pcl::PCLPointCloud2>;
+
+    //////////////////////////////////////////////////////////////////////////
+    /** \brief Concatenate two pcl::PCLPointCloud2.
+      * \param[in|out] cloud1 the first input and output point cloud dataset
+      * \param[in] cloud2 the second input point cloud dataset
+      * \return true if successful, false if failed (e.g., name/number of fields differs)
+      */
+    static bool
+    concatenate (pcl::PCLPointCloud2 &cloud1, const pcl::PCLPointCloud2 &cloud2);
+
+    /** \brief Concatenate two pcl::PCLPointCloud2.
+      * \param[in] cloud1 the first input point cloud dataset
+      * \param[in] cloud2 the second input point cloud dataset
+      * \param[out] cloud_out the resultant output point cloud dataset
+      * \return true if successful, false if failed (e.g., name/number of fields differs)
+      */
+    static bool
+    concatenate (const PCLPointCloud2 &cloud1,
+                 const PCLPointCloud2 &cloud2,
+                 PCLPointCloud2 &cloud_out)
+    {
+      cloud_out = cloud1;
+      return concatenate(cloud_out, cloud2);
+    }
+
+    /** \brief Add a point cloud to the current cloud.
+      * \param[in] rhs the cloud to add to the current cloud
+      * \return the new cloud as a concatenation of the current cloud and the new given cloud
+      */
+    inline PCLPointCloud2&
+    operator += (const PCLPointCloud2& rhs)
+    {
+      concatenate((*this), rhs);
```

I don't like that failures are silently ignored here. What about an exception?